includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1815,7 +1833,21 @@ public function add_contact( $contact, $list_id = false, $tags = [], $interests
 
 			// Create or update a list member.
 			$member_hash = Mailchimp::subscriberHash( $email_address );
-			$result      = $this->validate( $mc->put( "lists/$list_id/members/$member_hash", $update_payload ) );
+
+			/**
+			 * A default error message to show to readers if their signup request results in an error.
+			 *
+			 * @param string $reader_error The default error message.
+			 * @param string $email_address The email address that was attempted to be subscribed.
+			 * @param string $list_id The Mailchimp list ID that the email address was attempted to be subscribed to.
+			 */
+			$reader_error = apply_filters(
+				'newspack_newsletters_add_contact_reader_error_message',
+				__( "Sorry, this email cannot be subscribed to this newsletter. Please contact support with the email list you were trying to subscribe to and we'll add you to the list.", 'newspack-newsletters' ),
+				$email_address,
+				$list_id
+			);
```

Comment:
Nonblocking: This bit is repeated a few times in the code. Maybe we can move this into a helper method in the `Newspack_Newsletters_Service_Provider` class? (something like `get_add_contact_reader_error_message`)

Reply:
Great idea! I'll add this before merging and ping you for a re-review.

Reply:
@chickenn00dle 16edd2f implements a `get_add_contact_reader_error_message` method in the parent `Newspack_Newsletters_Service_Provider` class. It also consolidates additional params for the filter into a single arg, for more flexibility.